invenio_rdm_records/services/generators.py

```diff
@@ -262,3 +264,21 @@ def __init__(self, config_key, accept_values=[True], **kwargs):
     def _condition(self, **_):
         """Check if the config value is truthy."""
         return current_app.config.get(self.config_key) in self.accept_values
+
+
+class ResourceAccessToken(RecordOwners):
+    """Allow resource access tokens."""
+
+    def __init__(self, access):
+        """Constructor."""
+        self.access = access
+
+    def needs(self, record=None, file_key=None, **kwargs):
+        """Enabling Needs."""
+        record_owner = super().needs(record)
+
+        if record_owner and record and file_key:
+            signer_id = record_owner[0].value
+            return [RATNeed("files", signer_id, record["id"], file_key, self.access)]
```

Comment:
```suggestion
            return [RATNeed(signer_id, record["id"], file_key, self.access)]
```

minor: so that it's not file-specific (see also the comment on defining `RATNeed`). The `self.access` should be enough to define on what level we want to apply the permission.